app/javascript/components/widgets/twitter/tweet_content.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import Highlighter from 'react-highlight-words';
+import Microlink from '@microlink/react';
+import { colors, fonts, fontSizes } from '../../../lib/theme';
+
+const ContentWrapper = styled.div`
+  display: block;
+  width: auto;
+  height: 487px;
+  margin-bottom: 29px;
+`;
+
+const ContentImage = styled.img`
+  display: block;
+  width: auto;
+  height: 487px;
+  border-radius: 6px;
+  border-collapse: separate;
+`;
+
+const TweetText = styled.div`
+  color: #959292;
+  font-size: ${fontSizes.large};
+  margin-bottom: 29px;
+`;
+
+const getMentionsAndTags = text => {
+  const allWords = text.split(/[\s,\n]+/);
+  return allWords.filter(word => word.startsWith('@') || word.startsWith('#'));
+};
+
+export const TextOnly = displayText => (
+  <TweetText>
+    <Highlighter
+      highlightStyle={{
+        backgroundColor: 'transparent',
+        color: `${colors.white}`,
+      }}
+      searchWords={getMentionsAndTags(displayText)}
+      textToHighlight={displayText}
+    />
+  </TweetText>
+);
+
+export const BothMediaTypes = (displayText, mediaUrl, linkUrl) => (
+  <React.Fragment>
+    <div>
+      <TweetText>
+        <Highlighter
+          highlightStyle={{
+            backgroundColor: 'transparent',
+            color: `${colors.white}`,
+          }}
+          searchWords={getMentionsAndTags(displayText)}
+          textToHighlight={displayText}
+        />
+      </TweetText>
+      <TweetText
+        style={{
+          color: `${colors.white}`,
+          textDecoration: 'underline',
+        }}
+      >
+        {linkUrl}
+      </TweetText>
+    </div>
+    <TweetMedia mediaUrl={mediaUrl} linkUrl={undefined} />
+  </React.Fragment>
+);
+
+export const OneMediaType = (displayText, mediaUrl, linkUrl) => (
+  <React.Fragment>
+    <div>
+      <TweetText>
+        <Highlighter
+          highlightStyle={{
+            backgroundColor: 'transparent',
+            color: `${colors.white}`,
+          }}
+          searchWords={getMentionsAndTags(displayText)}
+          textToHighlight={displayText}
+        />
+      </TweetText>
+    </div>
+    <TweetMedia
+      mediaUrl={mediaUrl === null ? undefined : mediaUrl}
+      linkUrl={linkUrl === null ? undefined : linkUrl}
+    />
+  </React.Fragment>
+);
+
+const TweetMedia = ({ mediaUrl, linkUrl }) => (
+  <ContentWrapper>
+    {mediaUrl !== null ? <ContentImage src={mediaUrl} alt="twitter" /> : null}
```

Comment:
Prefer `condition && <Component />` to `condition ? <Component /> : null`. Was this a react native idiom?